mk/spksrc.patch.mk

```diff
@@ -17,14 +17,16 @@ endif
 
 # find patches into the following directory order:
 #    patches/*.patch
-#    patches/$(arch)/*.patch
 #    patches/DSM-$(TCVERSION)/*.patch
-#    patches/$(subarch)/*.patch
-#    patches/$(subarch)-$(TCVERSION)/*.patch
+#    patches/$(group)/*.patch
+#    patches/$(group)-$(TCVERSION)/*.patch
+#    patches/$(arch)/*.patch
+#    patches/$(arch)-$(TCVERSION)/*.patch
+# supported groups: arm5, arm7, arm7l, arm8, ppc, i686, x64
 ifeq ($(strip $(PATCHES)),)
-PATCHES = $(foreach arch,ARM5_ARCHES ARM7_ARCHES ARM8_ARCHES PPC_ARCHES x86_ARCHES x64_ARCHES, \
-	$(foreach subarch,$($(arch)), \
-	$(if $(filter $(ARCH),$(subarch)),$(sort $(wildcard patches/*.patch patches/DSM-$(TCVERSION)/*.patch patches/$(shell echo ${arch} | cut -f1 -d'_'| tr '[:upper:]' '[:lower:]')/*.patch patches/$(subarch)/*.patch patches/$(subarch)-$(TCVERSION)/*.patch)),)))
+PATCHES = $(foreach group,ARM5_ARCHS ARM7_ARCHS ARM7L_ARCHS ARM8_ARCHS PPC_ARCHS i686_ARCHS x64_ARCHS, \
+	$(foreach arch,$($(group)), \
+	$(if $(filter $(ARCH),$(arch)),$(sort $(wildcard patches/*.patch patches/DSM-$(TCVERSION)/*.patch patches/$(shell echo ${group} | cut -f1 -d'_'| tr '[:upper:]' '[:lower:]')/*.patch  patches/$(shell echo ${group} | cut -f1 -d'_'| tr '[:upper:]' '[:lower:]')-$(TCVERSION)/*.patch patches/$(arch)/*.patch patches/$(arch)-$(TCVERSION)/*.patch)),)))
```

Comment:
I remember having a hard time creating this 3x line code for patches...
besides the variable renaming (subarch,arch to arch,group), what is the effect of the change downstream?
If I get this right, this will allow managing patches at even a lower level such as `qoriq-6.1/*.patch`
Based on my previous comment related to ARM, perhaps then this could be renamed using `family` instead of `group` which would be more aligned.

Reply:
`subarch` was confusing me as it really means the arch. I use group to distinguish PPC, ARM5, ARM7, ARM8, x64 etc.. Family is already used to distiguish CPU families like intel core, intel atom etc. 
You already defined the lowest level like `qoriq-6.1/*.patch`, I only introduced a intermediate level like `ppc-5.2/*.patch`.

Reply:
But that might become quite useful for DSM7 migration later on.  I think it's a really good addition.